In [200]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import pyarrow

In [201]:
# Reading the clean data and make a copy of the original file to be the single source of truth
df = pd.read_parquet("cleaned_data.parquet", engine="pyarrow")
df.shape

(9993, 20)

In [202]:
df.isna().sum()

Order ID          0
Order Date        0
Ship Date         0
Ship Mode         0
Customer ID       0
Customer Name     0
Segment           0
Country/Region    0
City              0
State             0
Postal Code       0
Region            0
Product ID        0
Category          0
Sub-Category      0
Product Name      0
Sales             0
Quantity          0
Discount          0
Profit            0
dtype: int64

In [203]:
df.duplicated().sum()

np.int64(0)

In [204]:
df.describe()

,Order Date,Ship Date,Sales,Quantity,Discount,Profit
count,9993,9993,9993.000000,9993.000000,9993.000000,9993.000000
mean,2018-04-30 11:50:07.745421,2018-05-04 10:49:45.049534,229.852846,3.789753,0.156188,28.660971
min,2016-01-03 00:00:00,2016-01-07 00:00:00,0.444000,1.000000,0.000000,-6599.978000
25%,2017-05-23 00:00:00,2017-05-27 00:00:00,17.280000,2.000000,0.000000,1.731000
50%,2018-06-26 00:00:00,2018-06-29 00:00:00,54.480000,3.000000,0.200000,8.671000
75%,2019-05-14 00:00:00,2019-05-18 00:00:00,209.940000,5.000000,0.200000,29.364000
max,2019-12-30 00:00:00,2020-01-05 00:00:00,22638.480000,14.000000,0.800000,8399.976000
std,NaN,NaN,623.276074,2.225149,0.206457,234.271476


In [205]:
df.describe(include="str")

,Order ID,Customer ID,Customer Name,Country/Region,City,Postal Code,Product ID,Product Name
count,9993,9993,9993,9993,9993,9993,9993,9993
unique,5009,793,793,1,531,631,1862,1850
top,CA-2019-100111,WB-21850,William Brown,United States,New York City,10035.0,OFF-PA-10001970,Staple envelope
freq,14,37,37,9993,915,263,19,48


In [206]:
df.describe(include="category")

,Ship Mode,Segment,State,Region,Category,Sub-Category
count,9993,9993,9993,9993,9993,9993
unique,4,3,49,4,3,17
top,Standard Class,Consumer,California,West,Office Supplies,Binders
freq,5967,5191,2001,3203,6026,1523


In [207]:
df["Shipping Days"] = (df["Ship Date"] - df["Order Date"]).dt.days


In [208]:

df["Order Weekday"] = df["Order Date"].dt.day_name()
df["Order Month"] = df["Order Date"].dt.month
df["Order Quarter"] = df["Order Date"].dt.quarter
df["Order Year"] = df["Order Date"].dt.year
df["Is Weekend"] = df["Order Date"].dt.dayofweek >= 5
df[["Order Date", "Ship Date", "Shipping Days", "Order Weekday", "Is Weekend"]].head()

,Order Date,Ship Date,Shipping Days,Order Weekday,Is Weekend
0,2018-11-08,2018-11-11,3,Thursday,False
1,2018-11-08,2018-11-11,3,Thursday,False
2,2018-06-12,2018-06-16,4,Tuesday,False
3,2017-10-11,2017-10-18,7,Wednesday,False
4,2017-10-11,2017-10-18,7,Wednesday,False


In [209]:
df["Unit Price"] = df["Sales"] / df["Quantity"]
df["Profit Margin"] = (df["Profit"] / df["Sales"])*100
df[["Product ID", "Sales", "Unit Price", "Profit Margin"]].head()

,Product ID,Sales,Unit Price,Profit Margin
0,FUR-BO-10001798,261.9600,130.9800,16.00
1,FUR-CH-10000454,731.9400,243.9800,30.00
2,OFF-LA-10000240,14.6200,7.3100,47.00
3,FUR-TA-10000577,957.5775,191.5155,-40.00
4,OFF-ST-10000760,22.3680,11.1840,11.25


---

# بسم الله نبدأ 

---

# 📊 Business Questions

## 🎯 Objective
The goal of this analysis is to answer key business questions using data exploration, cleaning, and visualization techniques.  
Each question should be answered with appropriate Python code, tables, and visualizations where necessary.

---

# 🔟 Business Questions

### 1️⃣ Which product category generates the highest total sales?
> Identify the category with the highest revenue and compare it with the remaining categories.

---

### 2️⃣ Which product category generates the highest total profit?
> Determine the most profitable category and analyze whether high sales always lead to high profit.

---

### 3️⃣ Which customer segment contributes the most to overall sales?
> Compare customer segments and identify the one generating the highest revenue.

---

### 4️⃣ Which region achieves the highest sales and profit?
> Analyze regional performance to identify the strongest and weakest markets.

---

### 5️⃣ Which products are the top 10 best-selling products?
> Rank products based on total sales and visualize the top performers.

---

### 6️⃣ Does offering higher discounts increase sales or reduce profit?
> Investigate the relationship between discount, sales, and profit using statistics and visualizations.

---

### 7️⃣ Which shipping mode is used most frequently?
> Identify the most commonly used shipping method and compare its contribution to sales and profit.

---

### 8️⃣ How have sales changed over time?
> Analyze monthly and yearly sales trends to identify growth patterns or seasonal effects.

---

### 9️⃣ Which states or cities generate the highest revenue?
> Identify the top-performing geographic locations based on total sales and profit.

---

### 🔟 Are there products that generate high sales but low (or negative) profit?
> Detect products with strong sales but poor profitability and discuss possible business reasons.

---

# ✅ Deliverables

For each question:

- Perform the necessary data manipulation.
- Display summary tables.
- Create suitable visualizations.
- Write a short conclusion summarizing the findings.

---

> **Note:** Every answer should be supported by code, tables, and charts whenever appropriate.

---

### 📌 Question 1: Which product category generates the highest total sales?


In [210]:
top_category = (
    df.groupby("Category", observed=True)["Sales"]
      .sum()
      .sort_values(ascending=False)
)

top_category

Category
Technology         836154.0330
Furniture          741718.4233
Office Supplies    719047.0320
Name: Sales, dtype: float64

### 📌 Question 1: Which product category generates the highest total sales?

#### 📊 Results

| Rank | Category | Total Sales |
|------|-----------|------------:|
| 🥇 1 | Technology | **836,154.03** |
| 🥈 2 | Furniture | **741,718.42** |
| 🥉 3 | Office Supplies | **719,047.03** |

---

### 💡 Business Insight

- **Technology** is the highest revenue-generating category, with total sales of approximately **836K**.
- **Furniture** ranks second with approximately **742K** in total sales.
- **Office Supplies** comes third with approximately **719K** in sales.
- The results indicate that **Technology products are the primary driver of overall revenue**, making this category a strong candidate for continued investment in inventory, marketing, and promotions.

> **Conclusion:** Technology is the top-performing product category in terms of total sales and represents the largest revenue contributor to the business.

---

### 📌 Question 2: Which product category generates the highest total profit?


In [211]:
Product_Cat=df.groupby("Category", observed=True).agg(
    Total_Profit=("Profit","sum"),
    Average_Profit=("Profit","mean")
).sort_values("Total_Profit",ascending=False)

Product_Cat

,Total_Profit,Average_Profit
Category,,
Technology,145454.9481,78.752002
Office Supplies,122490.8008,20.327050
Furniture,18463.3316,8.709119


### 📌 Question 2: Which product category generates the highest total profit?

#### 📊 Results

| Rank | Category | Total Profit | Average Profit |
|------|-----------|-------------:|---------------:|
| 🥇 1 | Technology | **145,454.95** | **78.75** |
| 🥈 2 | Office Supplies | **122,490.80** | **20.33** |
| 🥉 3 | Furniture | **18,463.33** | **8.71** |

---

### 💡 Business Insight

- **Technology** is the most profitable category, generating approximately **145.45K** in total profit with the highest average profit per transaction (**78.75**).
- **Office Supplies** ranks second with approximately **122.49K** in total profit, although its average profit is significantly lower than Technology.
- **Furniture** produces the **lowest total and average profit**, despite having high sales in the previous analysis. This suggests that Furniture products may have **higher operating costs, lower profit margins, or larger discounts**.

---

### 📈 Comparison with Sales

- **Technology** ranks **first** in both **sales** and **profit**, making it the strongest-performing category.
- **Furniture** ranks **second in sales** but **last in profit**, indicating that high sales do not necessarily translate into high profitability.
- **Office Supplies** generates lower sales than Furniture but delivers **substantially higher profit**, making it a more profitable category overall.

> **Conclusion:** Technology is the most profitable product category and the primary contributor to business earnings. In contrast, Furniture requires further analysis to identify the factors reducing its profitability, such as discount strategies or product costs.

---

### 📌 Question 3: Which customer segment contributes the most to overall sales?


In [212]:
Customer_Segment = (
    df.groupby("Segment", observed=True)
      .agg(
          Total_Sales=("Sales", "sum"),
          Average_Sales=("Sales", "mean")
      )
      .sort_values("Total_Sales", ascending=False)
)

Customer_Segment

,Total_Sales,Average_Sales
Segment,,
Consumer,1.161401e+06,223.733644
Corporate,7.061464e+05,233.823300
Home Office,4.293718e+05,240.949370


### 📌 Question 3: Which customer segment contributes the most to overall sales?

#### 📊 Results

| Rank | Customer Segment | Total Sales | Average Sales |
|------|------------------|------------:|--------------:|
| 🥇 1 | Consumer | **1,161,400.83** | **223.73** |
| 🥈 2 | Corporate | **706,146.37** | **233.82** |
| 🥉 3 | Home Office | **429,371.84** | **240.95** |

---

### 💡 Business Insight

- **Consumer** is the largest revenue-generating customer segment, contributing approximately **1.16M** in total sales.
- **Corporate** ranks second with approximately **706K** in total sales, representing an important source of business revenue.
- **Home Office** contributes the lowest total sales (**429K**), but records the **highest average sales per order** (**240.95**), indicating that customers in this segment tend to place higher-value orders.

---

### 📈 Key Findings

- The **Consumer** segment is the primary driver of total revenue due to its large number of purchases.
- Although **Home Office** has the lowest overall sales, it generates the **highest average transaction value**, suggesting greater spending per order.
- **Corporate** balances both total sales and average order value, making it a valuable customer segment.

> **Conclusion:** The **Consumer** segment contributes the most to the company's overall sales, while the **Home Office** segment has the highest average sales per transaction. These insights can help guide marketing strategies, customer targeting, and promotional campaigns.

---

### 📌 Question 4: Which region achieves the highest sales and profit?


In [213]:
highest_region=df.groupby("Region",observed=True).agg(
     highest_Sales=("Sales","sum"), 
     Avrage_Sales=("Sales","mean"), 
     highest_Profit=("Profit","sum"), 
     Avrege_Profit=("Profit","mean") ).sort_values("highest_Sales", ascending=False)
highest_region

,highest_Sales,Avrage_Sales,highest_Profit,Avrege_Profit
Region,,,,
West,725457.8245,226.493233,108418.4489,33.849032
East,678499.8680,238.320993,91534.8388,32.151331
Central,501239.8908,215.772661,39706.3625,17.092709
South,391721.9050,241.803645,46749.4303,28.857673


### 📌 Question 4: Which region achieves the highest sales and profit?

#### 📊 Results

| Rank | Region | Total Sales | Average Sales | Total Profit | Average Profit |
|------|--------|------------:|--------------:|-------------:|---------------:|
| 🥇 1 | West | **725,457.82** | **226.49** | **108,418.45** | **33.85** |
| 🥈 2 | East | **678,499.87** | **238.32** | **91,534.84** | **32.15** |
| 🥉 3 | Central | **501,239.89** | **215.77** | **39,706.36** | **17.09** |
| 4 | South | **391,721.91** | **241.80** | **46,749.43** | **28.86** |

---

### 💡 Business Insight

- **West** is the best-performing region, generating the **highest total sales (725.46K)** and the **highest total profit (108.42K)**.
- **East** ranks second in both sales and profit, making it another strong contributor to overall business performance.
- **South** has the **highest average sales per order (241.80)** despite having the lowest total sales, indicating that customers in this region tend to place higher-value orders.
- **Central** records the **lowest average profit (17.09)** and relatively low total profit, suggesting lower profit margins or higher operating costs compared to other regions.

---

### 📈 Key Findings

- The **West** region is the company's strongest market in terms of both revenue and profitability.
- The **East** region maintains solid performance and represents a valuable growth opportunity.
- The **South** region shows strong customer spending per transaction but a smaller sales volume.
- The **Central** region requires further investigation to improve profitability and operational efficiency.

> **Conclusion:** The **West** region is the top-performing region, leading in both total sales and total profit. While **South** customers spend more on average per order, increasing transaction volume in this region could significantly boost overall business performance.

---

### 📌 Question 5: Which products are the top 10 best-selling products?


In [214]:
Products=df.groupby("Product Name",observed=True).agg(
Total_Sales=("Sales","sum"),
Average_Sales=("Sales","mean"),
Total_Quantity=("Quantity","sum"),
Total_Profit=("Profit","sum")
).sort_values("Total_Sales",ascending=False).head(10)
Products

,Total_Sales,Average_Sales,Total_Quantity,Total_Profit
Product Name,,,,
Canon imageCLASS 2200 Advanced Copier,61599.824,12319.964800,20,2.519993e+04
Fellowes PB500 Electric Punch Plastic Comb Binding Machine with Manual Bind,27453.384,2745.338400,31,7.753039e+03
Cisco TelePresence System EX90 Videoconferencing Unit,22638.480,22638.480000,6,-1.811078e+03
HON 5400 Series Task Chairs for Big and Tall,21870.576,2733.822000,39,3.979039e-13
GBC DocuBind TL300 Electric Binding System,19823.479,1802.134455,37,2.233505e+03
GBC Ibimaster 500 Manual ProClick Binding System,19024.500,2113.833333,48,7.609800e+02
Hewlett Packard LaserJet 3310 Copier,18839.686,2354.960750,38,6.983884e+03
"HP Designjet T520 Inkjet Large Format Printer - 24"" Color",18374.895,6124.965000,12,4.094977e+03
GBC DocuBind P400 Electric Binding System,17965.068,2994.178000,27,-1.878166e+03


### 📌 Question 5: Which products are the top 10 best-selling products?

#### 📊 Results

The analysis identified the **Top 10 products** based on their total sales revenue.

| Rank | Product | Total Sales | Average Sales | Total Quantity | Total Profit |
|------|---------|------------:|--------------:|---------------:|-------------:|
| 🥇 | Canon imageCLASS 2200 Advanced Copier | **61,599.82** | **12,319.96** | 20 | **25,199.93** |
| 🥈 | Fellowes PB500 Electric Punch Plastic Comb Binding Machine | **27,453.38** | **2,745.34** | 31 | **7,753.04** |
| 🥉 | Cisco TelePresence System EX90 Videoconferencing Unit | **22,638.48** | **22,638.48** | 6 | **-1,811.08** |
| 4 | HON 5400 Series Task Chairs for Big and Tall | **21,870.58** | **2,733.82** | 39 | **≈ 0.00** |
| 5 | GBC DocuBind TL300 Electric Binding System | **19,823.48** | **1,802.13** | 37 | **2,233.51** |
| 6 | GBC Ibimaster 500 Manual ProClick Binding System | **19,024.50** | **2,113.83** | 48 | **760.98** |
| 7 | Hewlett Packard LaserJet 3310 Copier | **18,839.69** | **2,354.96** | 38 | **6,983.88** |
| 8 | HP Designjet T520 Inkjet Large Format Printer | **18,374.90** | **6,124.97** | 12 | **4,094.98** |
| 9 | GBC DocuBind P400 Electric Binding System | **17,965.07** | **2,994.18** | 27 | **-1,878.17** |
| 10 | High Speed Automatic Electric Letter Opener | **17,030.31** | **5,676.77** | 11 | **-262.00** |

---

### 🔍 Business Insights

- The **Canon imageCLASS 2200 Advanced Copier** is the best-selling product, generating more than **61K** in total sales and over **25K** in profit, making it the company's most valuable product.
- Not every high-selling product is profitable. For example, the **Cisco TelePresence System EX90 Videoconferencing Unit** generated over **22K** in sales but resulted in a **net loss**.
- Some products, such as the **HON 5400 Series Task Chairs**, achieved strong sales while generating almost **no profit**, indicating very low profit margins.
- Several products in the top-selling list produced **negative profits**, suggesting that discounts, high costs, or pricing strategies may be reducing profitability.

---

### 📈 Recommendation

- Continue investing in products that achieve **both high sales and high profit**, such as the **Canon imageCLASS 2200 Advanced Copier**.
- Review the pricing and discount strategies for high-selling but **loss-making products** to improve profitability.
- Evaluate whether low-margin products should be repriced or supplied at lower costs to maximize business performance.

> **Conclusion:** High sales do not always guarantee high profitability. Business decisions should consider both **revenue and profit** when evaluating product performance.

---

### 📌 Question 6: Does offering higher discounts increase sales or reduce profit?


In [215]:
df["Discount_Level"]=pd.cut(df["Discount"],
    labels=["No Discount","Low Discount","Medium Discount","High Discount","Very High Discount"],
    bins=[-0.01,0,0.2,0.5,0.7,1])

In [216]:
Q6=df.groupby("Discount_Level",observed=True).agg(
    Total_Sales=("Sales","sum"),
    Average_Sales=("Sales","mean"),
    Total_Profit=("Profit","sum"),
    Average_Profit=("Profit","mean")
)
Q6

,Total_Sales,Average_Sales,Total_Profit,Average_Profit
Discount_Level,,,,
No Discount,1.087908e+06,226.742074,320987.6032,66.900292
Low Discount,8.465222e+05,222.593279,100785.4745,26.501571
Medium Discount,2.982600e+05,556.455298,-58804.9459,-109.710720
High Discount,4.726498e+04,85.008960,-46020.0121,-82.769806
Very High Discount,1.696376e+04,56.545853,-30539.0392,-101.796797


### 📌 Question 6: Does offering higher discounts increase sales or reduce profit?

#### 📊 Results

| Discount Level | Total Sales | Average Sales | Total Profit | Average Profit |
|---------------|------------:|--------------:|-------------:|---------------:|
| No Discount | **1,087,908.13** | **226.74** | **320,987.60** | **66.90** |
| Low Discount | **846,522.18** | **222.59** | **100,785.47** | **26.50** |
| Medium Discount | **298,260.03** | **556.46** | **-58,804.95** | **-109.71** |
| High Discount | **47,264.98** | **85.01** | **-46,020.01** | **-82.77** |
| Very High Discount | **16,963.76** | **56.55** | **-30,539.04** | **-101.80** |

---

### 💡 Business Insights

- Orders without discounts generated the **highest total sales (1.09M)** and the **highest total profit (320.99K)**.
- **Low Discounts** remained profitable, although both total and average profit were lower than orders without discounts.
- **Medium Discounts** recorded the **highest average sales per order (556.46)**, indicating that customers tend to place higher-value orders when moderate discounts are offered.
- However, **Medium, High, and Very High Discounts all resulted in negative profits**, suggesting that the additional revenue was not enough to offset the cost of the discounts.
- As the discount level increases, profitability declines significantly, eventually turning negative.

---

### 📈 Key Findings

- Moderate discounts may encourage larger purchases.
- Excessive discounts have a strong negative impact on profitability.
- **No Discount** and **Low Discount** achieve the best balance between revenue and profit.
- **Medium to Very High Discounts** should be carefully evaluated before being applied.

> **Conclusion:** The analysis suggests that while moderate discounts may increase the average order value, higher discount levels consistently reduce profitability. The company should optimize its discount strategy to maximize both revenue and profit rather than relying on deep discounts.

---

### 📌 Question 7: Which shipping mode is used most frequently?


In [217]:
Top_Shipping_Mode=df.groupby("Ship Mode",observed=True).agg(
Number_of_Orders=("Order ID","count")
).sort_values("Number_of_Orders",ascending=False)
Top_Shipping_Mode

,Number_of_Orders
Ship Mode,
Standard Class,5967
Second Class,1945
First Class,1538
Same Day,543


### 📌 Question 7: Which shipping mode is used most frequently?

#### 📊 Results

| Rank | Shipping Mode | Number of Orders |
|------|---------------|-----------------:|
| 🥇 1 | Standard Class | **5,967** |
| 🥈 2 | Second Class | **1,945** |
| 🥉 3 | First Class | **1,538** |
| 4 | Same Day | **543** |

---

### 💡 Business Insights

- **Standard Class** is the most frequently used shipping method, accounting for **5,967 orders**, making it the preferred shipping option for customers.
- **Second Class** ranks second with **1,945 orders**, followed by **First Class** with **1,538 orders**.
- **Same Day** shipping is the least frequently used option, with only **543 orders**, indicating limited demand for expedited delivery.

---

### 📈 Key Findings

- Most customers prefer **Standard Class**, likely because it provides a balance between delivery time and shipping cost.
- Premium shipping options (**First Class** and **Same Day**) are selected much less frequently, suggesting that customers may prioritize lower shipping costs over faster delivery.
- The company should ensure that **Standard Class** operations remain efficient, as it handles the majority of shipments.

> **Conclusion:** **Standard Class** is by far the dominant shipping mode, representing the largest share of customer orders. Optimizing this shipping method can have the greatest impact on customer satisfaction and operational efficiency.

---

### 📌 Question 8: How have sales changed over time?


In [218]:
Sales_Year=df.groupby("Order Year",observed=True).agg(
    Total_Sales=("Sales","sum"),
    Avrage_Sales=("Sales","mean")
)
Sales_Year

,Total_Sales,Avrage_Sales
Order Year,,
2016,483966.1261,242.954883
2017,470532.5090,223.849909
2018,609205.5980,235.487282
2019,733215.2552,221.381418


In [219]:
Sales_Month=df.groupby("Order Month",observed=True).agg(
    Total_Sales=("Sales","sum"),
    Avrage_Sales=("Sales","mean")
)
Sales_Month

,Total_Sales,Avrage_Sales
Order Month,,
1,94924.8356,249.146550
2,59751.2514,199.170838
3,205005.4888,294.548116
4,137480.7566,206.118076
5,155028.8117,210.923553
6,152718.6793,212.996763
7,147238.0970,207.377601
8,159044.0630,225.274877
9,307649.9457,222.451154


### 📌 Question 8: How have sales changed over time?

## 📊 Sales by Year

| Year | Total Sales | Average Sales |
|-----:|------------:|--------------:|
| 2016 | **483,966.13** | **242.95** |
| 2017 | **470,532.51** | **223.85** |
| 2018 | **609,205.60** | **235.49** |
| 2019 | **733,215.26** | **221.38** |

### 💡 Yearly Insights

- Sales experienced a slight decline from **2016** to **2017**.
- The business recovered strongly in **2018**, with total sales increasing significantly.
- **2019** recorded the **highest annual sales (733.22K)**, indicating continued business growth.
- Although the average order value fluctuated over the years, total revenue showed a clear upward trend after 2017.

---

## 📊 Sales by Month

| Month | Total Sales | Average Sales |
|------:|------------:|--------------:|
| January | **94,924.84** | **249.15** |
| February | **59,751.25** | **199.17** |
| March | **205,005.49** | **294.55** |
| April | **137,480.76** | **206.12** |
| May | **155,028.81** | **210.92** |
| June | **152,718.68** | **213.00** |
| July | **147,238.10** | **207.38** |
| August | **159,044.06** | **225.27** |
| September | **307,649.95** | **222.45** |
| October | **200,322.98** | **244.59** |
| November | **352,461.07** | **239.61** |
| December | **325,293.50** | **231.03** |

### 💡 Monthly Insights

- **November** generated the **highest monthly sales (352.46K)**.
- **December** ranked second with **325.29K**, followed by **September (307.65K)**.
- **February** recorded the **lowest sales (59.75K)** during the year.
- Sales increased noticeably during the **last quarter (September–December)**, suggesting strong seasonal demand.
- **March** achieved the **highest average sales per order (294.55)**, indicating larger order values during that month.

---

## 📈 Key Findings

- The company demonstrates a **positive long-term growth trend**, with annual sales peaking in **2019**.
- Sales are **seasonal**, with the strongest performance occurring between **September and December**, especially in **November**.
- The first half of the year generally generates lower revenue than the second half.

> **Conclusion:** Sales have grown steadily over the four-year period, while monthly analysis reveals a clear seasonal pattern. The business performs best during the final months of the year, making this period the most important for marketing campaigns, inventory planning, and sales operations.

---

### 📌 Question 9: Which states generate the highest revenue?


In [220]:
Rank=df.groupby("State",observed=True).agg(
    Total_Sales=("Sales","sum"),
    Average_Salse=("Sales","mean"),
    Total_profit=("Profit","sum")
).sort_values("Total_Sales",ascending=False).head(10)
Rank

,Total_Sales,Average_Salse,Total_profit
State,,,
California,457687.6315,228.729451,76381.3871
New York,310876.2710,275.599531,74038.5486
Texas,170188.0458,172.779742,-25729.3563
Washington,138641.2700,273.994605,33402.6517
Pennsylvania,116511.9140,198.487077,-15559.9603
Florida,89473.7080,233.612815,-3399.3017
Illinois,80166.1010,162.939230,-12607.8870
Ohio,77976.7640,166.617017,-16959.3178
Michigan,76269.6140,299.096525,24463.1876


### 📌 Question 9: Which states generate the highest revenue?

#### 📊 Results

| Rank | State | Total Sales | Average Sales | Total Profit |
|------|--------|------------:|--------------:|-------------:|
| 🥇 1 | California | **457,687.63** | **228.73** | **76,381.39** |
| 🥈 2 | New York | **310,876.27** | **275.60** | **74,038.55** |
| 🥉 3 | Texas | **170,188.05** | **172.78** | **-25,729.36** |
| 4 | Washington | **138,641.27** | **273.99** | **33,402.65** |
| 5 | Pennsylvania | **116,511.91** | **198.49** | **-15,559.96** |
| 6 | Florida | **89,473.71** | **233.61** | **-3,399.30** |
| 7 | Illinois | **80,166.10** | **162.94** | **-12,607.89** |
| 8 | Ohio | **77,976.76** | **166.62** | **-16,959.32** |
| 9 | Michigan | **76,269.61** | **299.10** | **24,463.19** |
| 10 | Virginia | **70,636.72** | **315.34** | **18,597.95** |

---

### 💡 Business Insights

- **California** is the highest revenue-generating state, with **457.69K** in total sales and the highest overall profit (**76.38K**).
- **New York** ranks second in both sales and profit, making it another key market for the business.
- **Texas** is the **third-highest revenue** state but records a **significant financial loss (-25.73K)**, indicating poor profitability despite strong sales.
- **Washington**, **Michigan**, and **Virginia** maintain positive profits while generating solid sales, making them attractive markets for future growth.
- Several high-revenue states, including **Pennsylvania**, **Florida**, **Illinois**, and **Ohio**, generate **negative profits**, suggesting that high revenue does not always translate into profitability.

---

### 📈 Key Findings

- Revenue is highly concentrated in **California** and **New York**, which are also the company's most profitable states.
- Some states generate strong sales but suffer from negative profit margins, highlighting potential issues with pricing, discounts, or operating costs.
- Profitability should be analyzed alongside revenue when evaluating market performance.

> **Conclusion:** **California** is the company's strongest market, leading in both sales and profit. However, states such as **Texas** and **Pennsylvania** require further investigation to understand why high sales are accompanied by financial losses.

---

### 📌 Question 10: Are there products that generate high sales but low (or negative) profit?


In [221]:
Q10 = (
    df.groupby("Product Name", observed=True)
      .agg(
          Total_Sales=("Sales", "sum"),
          Total_Quantity=("Quantity", "sum"),
          Total_Profit=("Profit", "sum")
      )
      .query("Total_Profit <= 0")
      .sort_values("Total_Sales", ascending=False)
      .head(10)
)

Q10

,Total_Sales,Total_Quantity,Total_Profit
Product Name,,,
Cisco TelePresence System EX90 Videoconferencing Unit,22638.4800,6,-1811.0784
GBC DocuBind P400 Electric Binding System,17965.0680,27,-1878.1662
High Speed Automatic Electric Letter Opener,17030.3120,11,-262.0048
Lexmark MX611dhe Monochrome Laser Printer,16829.9010,18,-4589.9730
Martin Yale Chadless Opener Electric Letter Opener,16656.2000,22,-1299.1836
"Riverside Palais Royal Lawyers Bookcase, Royale Cherry Finish",15610.9656,24,-669.5448
Bretford Rectangular Conference Table Tops,12995.2915,46,-327.2331
Cubify CubeX 3D Printer Double Head Print,11099.9630,9,-8879.9704
Chromcraft Bull-Nose Wood Oval Conference Tables & Bases,9917.6400,27,-2876.1156


### 📌 Question 10: Are there products that generate high sales but low (or negative) profit?

#### 📊 Results

| Rank | Product | Total Sales | Total Profit | Total Quantity |
|------|---------|------------:|-------------:|---------------:|
| 1 | Cisco TelePresence System EX90 Videoconferencing Unit | **22,638.48** | **-1,811.08** | 6 |
| 2 | GBC DocuBind P400 Electric Binding System | **17,965.07** | **-1,878.17** | 27 |
| 3 | High Speed Automatic Electric Letter Opener | **17,030.31** | **-262.00** | 11 |
| 4 | Lexmark MX611dhe Monochrome Laser Printer | **16,829.90** | **-4,589.97** | 18 |
| 5 | Martin Yale Chadless Opener Electric Letter Opener | **16,656.20** | **-1,299.18** | 22 |
| 6 | Riverside Palais Royal Lawyers Bookcase, Royale Cherry Finish | **15,610.97** | **-669.54** | 24 |
| 7 | Bretford Rectangular Conference Table Tops | **12,995.29** | **-327.23** | 46 |
| 8 | Cubify CubeX 3D Printer Double Head Print | **11,099.96** | **-8,879.97** | 9 |
| 9 | Chromcraft Bull-Nose Wood Oval Conference Tables & Bases | **9,917.64** | **-2,876.12** | 27 |
| 10 | Bush Advantage Collection Racetrack Conference Table | **9,544.73** | **-1,934.40** | 33 |

---

### 💡 Business Insights

- Several products generate **high sales revenue but negative profit**, indicating that strong sales do not necessarily translate into business success.
- The **Cisco TelePresence System EX90 Videoconferencing Unit** is the highest-selling product in this group, yet it still results in a financial loss.
- The **Cubify CubeX 3D Printer Double Head Print** records one of the **largest losses (-8.88K)** despite generating more than **11K** in sales, suggesting serious profitability issues.
- Most products in this list are expensive items with relatively low sales quantities, which may indicate high production costs, aggressive discounting, or pricing strategies that reduce profitability.

---

### 📈 Key Findings

- High revenue alone is **not a reliable indicator of business performance**.
- Products with persistent negative profits should be reviewed to identify the causes of their losses.
- Pricing strategies, discount policies, supplier costs, and operational expenses should be evaluated before continuing to promote these products.

---

### 📌 Recommendation

- Review the pricing and discount policies for these products.
- Investigate procurement and production costs to identify opportunities for improving margins.
- Consider discontinuing or redesigning products that consistently generate losses unless they provide strategic value, such as attracting customers or driving sales of complementary products.

> **Conclusion:** The analysis reveals that several high-selling products are **unprofitable**, emphasizing the importance of evaluating both **sales and profit** when measuring product performance. Focusing only on revenue could lead to decisions that negatively impact the company's overall profitability.

---

In [222]:
df.to_parquet("cleaned_data.parquet",engine="pyarrow")